### DISCUSSION TITLE
Is there a Svelte version?

### DISCUSSION THREAD

**Discussion by frederikhors** [radix-ui/primitives#1138] [Help] [answered]
Is there a Svelte version?

**Comment by jjenzz** [comment, marked as answer]
We're focusing on React only atm I'm afraid.

**Comment by jyoungblood** [comment]
I think Radix is phenomenal and would absolutely LOVE to see a Svelte version. Since this isn't really in the cards, it seems to be a perfect opportunity for someone to build something similar for Svelte.

I'm only one (currently very time-constrained) person, though, so it's unlikely that I'll be the guy to build it...but I would love to collaborate with other people who want to make this a reality.

So if you ended up at this thread from a "Radix for Svelte" Google search and feel inspired to build a high-quality system like this , please get in touch!

**Reply by LexSwed** [reply]
I think all, Svelte, Vue and Solid have a decent Web Components support. I understand that building a web components library is basically rebuilding something that has to be provided by the browser, but there are good examples like [Spectrum](https://opensource.adobe.com/spectrum-web-components/) where Web Components show great value.

**Comment by noname-oni** [comment]
Svelte for Radix ✊

**Comment by Nisthar** [comment]
> Is there a Svelte version?

Just saw primitives and was thinking the same lol.

**Reply by noname-oni** [reply]
Absolutely same

**Comment by zakaria-chahboun** [comment]
Svelte 🔥

**Comment by dit7ya** [comment]
Looking forward to a Svelte port.

**Comment by noname-oni** [comment]
I think skeleton do something similar to radix

**Comment by yekta** [comment]
It would be awesome to see a Svelte port. This is definitely one of the things you miss when using Svelte.

**Comment by 3goats** [comment]
Plus 1 for a Svelte port

**Reply by zakaria-chahboun** [reply]
https://www.skeleton.dev/

**Reply by noname-oni** [reply]
> https://www.skeleton.dev/

Skeleton is good 👍

**Reply by yekta** [reply]
> https://www.skeleton.dev/

![9DA82BB3-F4FC-446F-809C-DADCB6E8D74A](https://user-images.githubusercontent.com/40706601/213329165-94f1f00d-f101-40de-8319-3a40992a47af.png)

For me a big selling point of Radix is how refined it is. Dropdowns etc. being “aware” of where they are placed for example comes in very handy, or type ahead or multilevel dropdowns that work exactly like you would expect them to without unintentional closing. I find it particularly time consuming to build a bullet proof versions of those myself.

Looking at skeleton’s own theme dropdown, it doesn’t seem aware.

**Comment by TGlide** [comment]
Hi! We're working on a Radix port, still early days, but contributors are welcome 😄 
https://github.com/TGlide/radix-svelte

**Reply by TGlide** [reply]
I have named the project radix-svelte, but if anyone from the team feels that they don't want the radix name associated with it, please let me know. Pinging @jjenzz and @benoitgrelard

**Comment by diegoulloao** [comment]
Ok is 2024. Svelte needed.

**Reply by noname-oni** [reply]
100%

**Reply by TGlide** [reply]
Several projects exist that bring similar functionality to Svelte. Bits UI, Melt UI, shadcn-svelte, etc. 

I tried porting Radix to Svelte, but didn't have much success. You can check out what I did here: https://github.com/TGlide/radix-svelte
